src/index.ts

```diff
@@ -159,70 +163,84 @@ export class HttpClient implements IHttpClient {
     return error;
   }
 
+
+
   async get<T>(path: string, options?: RequestOptions): Promise<T> {
+    const o = this.#processOptions( 'GET', undefined, options);
     return this.#requestJson({
       path: path,
-      options: { ...options, method: 'GET' }
+      options: o
     });
   }
-  async post<T>(path: string, options?: RequestOptions): Promise<T> {
+  async post<T>(path: string, data?: Record<string, unknown>, options?: Omit<RequestOptions, 'json'>): Promise<T> {
+    const o = this.#processOptions( 'POST', data, options);
     return this.#requestJson({
       path: path,
-      options: { ...options, method: 'POST' }
+      options: o
     });
   }
-  async put<T>(path: string, options?: RequestOptions): Promise<T> {
+  async put<T>(path: string, data?: Record<string, unknown>, options?: RequestOptions): Promise<T> {
+    const o = this.#processOptions( 'PUT', data, options)
     return this.#requestJson({
       path: path,
-      options: { ...options, method: 'PUT'}
+      options: o
     });
   }
-  async patch<T>(path: string, options?: RequestOptions): Promise<T> {
+  async patch<T>(path: string, data?: Record<string, unknown>, options?: RequestOptions): Promise<T> {
+    const o = this.#processOptions( 'PATCH', data, options)
     return this.#requestJson({
       path: path,
-      options: { ...options, method: 'PATCH'}
+      options: o
     });
   }
   async delete(path: string, options?: RequestOptions): Promise<void> {
+    const o = this.#processOptions( 'DELETE', undefined, options)
     await this.#requestJson({
       path: path,
-      options: { ...options, method: 'DELETE'}
+      options: o
     });
     return undefined;
   }
-  async request(path: string, options: RequestOptions): Promise<Response<unknown>> {
+  /**
+   * @description Access to all 'got' supported options, returns the response instead of JSON.
+   * @param {string} path path or url
+   * @param {RequestOptions} options all options supported by 'got'
+   * @return {Response<unknown>}
+   */
+  async raw(path: string, options: RequestOptions): Promise<Response<unknown>> {
       return this.got(path, options)
   }
 }
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-type HttpMethod = 'get' | 'post' | 'patch' | 'put' | 'delete';
+type HttpMethodCall = 'get' | 'post' | 'patch' | 'put' | 'delete';
 export class TestHttpClient implements IHttpClient {
   public _routePayloads: Record<string,Record<string, any>>
-  public calls: Record<HttpMethod, Record<string, any>>
+  public calls: Record<HttpMethodCall, Record<string, any>>
```

Comment:
Fields are public per default, so you don't really need that keyword. Idiomatic TS usually omits it.

Reply:
Yeah sorry, copy paste, didn't see it